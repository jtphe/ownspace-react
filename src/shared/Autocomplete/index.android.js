import React from 'react';
import Autocomplete from 'react-native-autocomplete-input';
import { Text, View, StyleSheet, Keyboard } from 'react-native';
import i18n from '@i18n/i18n';
import { TouchableOpacity } from 'react-native-gesture-handler';

/**
 * The autoCompleteAndroid component
 * @param {Object[]} users - The users of the group
 * @param {string} query - The query typed by the user
 * @param {function} setQuery - The function that set the query typed by the user
 * @param {function} comp - Compare the query typed by the user with the first user in the list
 */
const autoCompleteAndroid = ({
  users,
  query,
  setQuery,
  comp,
  addGuestsList
}) => {
  /**
   * Render the autoCompleteAndroid class component
   * @returns {React.Component} - autoCompleteAndroid component
   */
  return (
    <View style={styles.autocompleteContainer}>
      <Autocomplete
        data={
          users.length === 1 &&
          comp(
            query,
            users[0].firstname !== null ? users[0].firstname : users[0].email
          )
            ? []
            : users
        }
        defaultValue={query}
        style={styles.inputInvitations}
        inputContainerStyle={styles.autocompleteInputContainer}
        listContainerStyle={styles.listContainerStyle}
        placeholderTextColor="rgba(0, 0, 0, 0.5)"
        placeholder={i18n.t('shareModal.placeholder')}
        onSubmitEditing={guest => addGuestsList(guest.nativeEvent.text)}
        onChangeText={text => setQuery(text)}
        renderItem={({ item }) => (
          <TouchableOpacity
            onPress={() => {
              addGuestsList(item);
              Keyboard.dismiss();
            }}
          >
            {item.firstname !== null && item.lastname !== null ? (
              <Text style={styles.userItem}>
                {`${item.firstname} ${item.lastname}`}
              </Text>
            ) : (
              <Text style={styles.userItem}>{`${item.email}`}</Text>
            )}
          </TouchableOpacity>
        )}
      />
    </View>
  );
};

/**
 * Styles of autoCompleteAndroid component
 */
const styles = StyleSheet.create({
  autocompleteContainer: {
    left: 0,
    position: 'absolute',
    top: 0,
    zIndex: 100
  },
  inputInvitations: {
    color: '#555',
    fontSize: 16,
    height: 45,
    width: 230,
    marginLeft: 10
  },
  autocompleteInputContainer: {
    borderRadius: 4,
    borderWidth: 1,
    borderColor: '#ccc'
  },
  listContainerStyle: {
    width: '108%',
    marginBottom: 20,
    marginLeft: -10,
    zIndex: 100
  },
  userItem: {
    color: '#555',
    fontSize: 16,
    height: 30,
    marginLeft: 10,
    marginTop: 10
  }
});

autoCompleteAndroid.defaultProps = {
  users: [],
  query: ''
};

export default autoCompleteAndroid;
