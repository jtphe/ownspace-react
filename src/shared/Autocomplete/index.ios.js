import React from 'react';
import Autocomplete from 'react-native-autocomplete-input';
import { StyleSheet, TouchableOpacity } from 'react-native';
import Text from '@shared/Text';
import i18n from '@i18n/i18n';

/**
 * The autoCompleteIos component
 * @param {Object[]} users - The users of the group
 * @param {string} query - The query typed by the user
 * @param {function} setQuery - The function that set the query typed by the user
 * @param {function} comp - Compare the query typed by the user with the first user in the list
 */
const autoCompleteIos = ({ users, query, setQuery, comp, addGuestsList }) => {
  /**
   * Render the autoCompleteIos class component
   * @returns {React.Component} - autoCompleteIos component
   */
  return (
    <Autocomplete
      data={
        users.length === 1 &&
        comp(
          query,
          users[0].firstname !== null ? users[0].firstname : users[0].email
        )
          ? []
          : users
      }
      defaultValue={query}
      style={styles.inputInvitations}
      inputContainerStyle={styles.autocompleteInputContainer}
      listContainerStyle={styles.listContainerStyle}
      keyboardShouldPersistTaps="always"
      placeholderTextColor="rgba(0, 0, 0, 0.5)"
      placeholder={i18n.t('shareModal.placeholder')}
      onSubmitEditing={guest => addGuestsList(guest.nativeEvent.text)}
      onChangeText={text => setQuery(text)}
      renderItem={({ item }) => (
        <TouchableOpacity onPress={() => addGuestsList(item)}>
          {item.firstname !== null && item.lastname !== null ? (
            <Text style={styles.userItem}>
              {`${item.firstname} ${item.lastname}`}
            </Text>
          ) : (
            <Text style={styles.userItem}>{`${item.email}`}</Text>
          )}
        </TouchableOpacity>
      )}
    />
  );
};

/**
 * Styles of autoCompleteIos component
 */
const styles = StyleSheet.create({
  inputInvitations: {
    color: '#555',
    fontSize: 16,
    height: 45,
    width: 190,
    marginLeft: 10
  },
  autocompleteInputContainer: {
    borderRadius: 4,
    borderWidth: 1,
    borderColor: '#ccc'
  },
  listContainerStyle: {
    width: '100%',
    marginBottom: 20
  },
  userItem: {
    color: '#555',
    fontSize: 16,
    height: 30,
    marginLeft: 10,
    marginTop: 10
  }
});

autoCompleteIos.defaultProps = {
  users: [],
  query: ''
};

export default autoCompleteIos;
