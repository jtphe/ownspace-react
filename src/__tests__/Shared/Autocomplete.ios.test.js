import React from 'react';
import Autocomplete from '@shared/Autocomplete/index.ios';
import renderer from 'react-test-renderer';
import * as Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

describe('Autocomplete ios tests', () => {
  it('Autocomplete snapshot', () => {
    const tree = renderer.create(<Autocomplete />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

Enzyme.configure({ adapter: new Adapter() });
