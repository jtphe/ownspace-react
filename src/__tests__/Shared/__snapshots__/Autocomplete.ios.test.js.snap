// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete ios tests Autocomplete snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "zIndex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "borderColor": "#b9b9b9",
          "borderRadius": 1,
          "borderWidth": 1,
        },
        Object {
          "borderColor": "#ccc",
          "borderRadius": 4,
          "borderWidth": 1,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      data={Array []}
      defaultValue=""
      flatListProps={Object {}}
      inputContainerStyle={
        Object {
          "borderColor": "#ccc",
          "borderRadius": 4,
          "borderWidth": 1,
        }
      }
      keyboardShouldPersistTaps="always"
      listContainerStyle={
        Object {
          "marginBottom": 20,
          "width": "100%",
        }
      }
      onChangeText={[Function]}
      onEndEditing={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Enter a user name..."
      placeholderTextColor="rgba(0, 0, 0, 0.5)"
      rejectResponderTermination={true}
      renderItem={[Function]}
      renderSeparator={null}
      renderTextInput={[Function]}
      style={
        Object {
          "color": "#555",
          "fontSize": 16,
          "height": 45,
          "marginLeft": 10,
          "width": 190,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "marginBottom": 20,
        "width": "100%",
      }
    }
  />
</View>
`;
